hal/src/rtl872x/dct_hal.cpp

```diff
@@ -138,8 +138,8 @@ class DcdFile {
 
         int flags = LFS_O_RDWR;
 
-        if (r) {
-            /* Create as well */
+        if (r || info.size == 0) {
```

Comment:
Could we also check for `!= or < an expected size` in case a device is reset in the middle of the 0xFF writing loop?

Reply:
`info.size < sizeof(application_dct_t)` ?

Reply:
That's guaranteed to either complete entirely or not at all. Also basing things on size is going to cause problems if we ever have to extend `application_dct_t`. We'll just jump straight on this landmine.

Reply:
Lemme run your PR test... and I'll try to reset mid-0xFF write and see if that size is still 0.  If it's guaranteed, I'm good with that.

Reply:
(If it wasn't, we'd have all sorts of consistency issues everywhere not just DCT)